src/distributions/microsoft/installer.ts

```diff
@@ -26,10 +26,18 @@ export class MicrosoftDistributions extends JavaBase {
     core.info(
       `Downloading Java ${javaRelease.version} (${this.distribution}) from ${javaRelease.url} ...`
     );
-    const javaArchivePath = await tc.downloadTool(javaRelease.url);
+    let javaArchivePath = await tc.downloadTool(javaRelease.url);
 
     core.info(`Extracting Java archive...`);
     const extension = getDownloadArchiveExtension();
+    if (
+      process.platform === 'win32' &&
+      (this.architecture === 'arm64' || this.architecture === 'aarch64')
+    ) {
+      const javaArchivePathRenamed = `${javaArchivePath}.zip`;
+      await fs.renameSync(javaArchivePath, javaArchivePathRenamed);
+      javaArchivePath = javaArchivePathRenamed;
+    }
```

Comment:
Nit: We could add comment, clarifying why we need to append `.zip` to the file name.

Reply:
Done. Thank you.